### PR TITLE
Reduce boilerplate in plugins utils tests

### PR DESCRIPTION
**Pull request by rjernst**
Tests for PluginsUtils need to create many different plugin descriptors.
But the vast majority of these just need a plugin name and dependencies.
This commit adds a utility method to simplify these tests.

### CODE REVIEW

**Review by jdconrad — APPROVED** [review]
LGTM

File: server/src/test/java/org/elasticsearch/plugins/PluginsUtilsTests.java [hunk 1]
```diff
@@ -38,6 +37,21 @@
 @LuceneTestCase.SuppressFileSystems(value = "ExtrasFS")
 public class PluginsUtilsTests extends ESTestCase {
 
+    PluginDescriptor newTestDescriptor(String name, List<String> deps) {
```
Comment (stu-elastic) [inline server/src/test/java/org/elasticsearch/plugins/PluginsUtilsTests.java:40]: Maybe make deps variadic to avoid list creation by caller?

**Review by stu-elastic — APPROVED** [review]
lgtm, minor suggestion.

File: server/src/test/java/org/elasticsearch/plugins/PluginsUtilsTests.java (hunk 1 above)
Comment (rjernst) [reply server/src/test/java/org/elasticsearch/plugins/PluginsUtilsTests.java:40]: We have to create a List for the descriptor regardless.

**Pull request merged**